src/relax/op/op.cc

```diff
@@ -177,7 +227,9 @@ TVM_REGISTER_OP("relax.call_builtin_with_ctx")
     .set_num_inputs(4)
     .add_argument("func", "Expr", "The builtin packed func.")
     .add_argument("args", "Tuple", "The input arguments.")
-    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoCallBuiltinWithCtx);
+    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoCallBuiltinWithCtx)
+    // TODO(relax-team): Please verify if these are normally impure or not
```

Comment:
seems the existing builtin packedfuncs are pure

Reply:
Are we sure all of them are?

Reply:
I noticed that there are builtins for updating a key-value store (those related to the `attention_kv_cache`). Updates are not pure, so that's a counterexample